### ISSUE TITLE
Inconsistency in documentation

### ISSUE THREAD

**Issue by floriancastelain** [ibis-project/ibis#3514]
In [Impala backend](https://ibis-project.org/backends/Impala/), in Connection API>method>do_connect:

- The method listed is called `do_connect`
- However, the examples use `connect`
- I ahve always used `connect`

Question is: Is it then `connect` or `do_connect` ? The incorrect one should be ajusted in the documentation. If both of them are available and doing the same thing, then why is this redundant stuff in here ? Moreover, it should be stated somewhere.

**Comment by cpcloud** [comment]
@floriancastelain Thanks for bringing this up.

TL; DR: the user-facing API is `ibis.${backend}.connect` which maps to `do_connect`. `do_connect` shows up in the docs because that's where the explicit connection parameters are listed.

What's currently happening is this:

1. Accessing `ibis.${backend}` loads a entry point for the backend, which is `ibis.backends.${backend}.Backend`.
2. The `Backend` class is constructed and the instance is set to `ibis.${backend}`, effectively `ibis.${backend} = Backend()`.
3. `ibis.${backend}.connect` is actually the `connect` method on an instance of the `Backend` class.
4. `connect` constructs the `Backend` with the backend specific connection parameters (this seems redundant, since we're instantiating `Backend` upon attribute access and then again in the `connect` invocation).

`connect`'s signature is `*args, **kwargs` which makes it all but useless for documentation.

Perhaps in the short term we can say something like

> `connect` is the user facing API. See the backend's `do_connect` method for connection parameter details.

with proper links etc.
